src/Functional.CQS/IAsyncCommandHandler.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace Functional.CQS
+{
+	/// <summary>
+	/// Interface for asynchronous command handlers.  Intended to encapsulate application logic that changes the state of the system and that can potentially fail.
+	/// </summary>
+	/// <typeparam name="TCommand">The command type.</typeparam>
+	/// <typeparam name="TError">The error type.</typeparam>
+	public interface IAsyncCommandHandler<in TCommand, TError>
+		where TCommand : ICommandParameters<TError>
+	{
+		/// <summary>
+		/// Execute the command.
+		/// </summary>
+		/// <param name="parameters">The command parameters.</param>
+		/// <param name="cancellationToken">The cancellation token.</param>
+		/// <returns></returns>
+		Task<Result<Unit, TError>> HandleAsync(TCommand parameters, CancellationToken cancellationToken);
```

Comment:
Based on conversations that I've had internally with people some time ago, I decided to force command handlers to return `Result<Unit, TError>` rather than `void`.  Pushes people toward better pattern usage.

Reply:
After talking about this in Slack and reviewing the `IQ.Vanilla` issue, I agree with this approach. Mike raised a good point in the ticket, and I don't really see a scenario where I would prefer getting a `void` over a `Result<Unit, TError>`.